src/main/java/org/passay/PasswordData.java

```diff
@@ -14,6 +14,16 @@
 public class PasswordData
 {
 
+  /** Enum to define the origin of a password. */
+  public enum Origin {
+
+    /** a password which was generated by a typical human user. */
+    USER_GENERATED,
```

Comment:
Not crazy about these names. The casing is wrong and I would think simply "User" and "Generated" would suffice. You really want to capture whether the user supplied the password or not, and I think that's sufficiently clear with my suggestions.